pleskdistup/actions/common_checks.py

```diff
@@ -98,150 +98,229 @@ def _do_check(self) -> bool:
+    target_versions: typing.List[version.PHPVersion]
 
     def __init__(
         self,
-        min_version: str,
+        target_versions: typing.List[str],
     ):
-        self.name = "check for outdated PHP versions"
-        self.min_version = version.PHPVersion(min_version)
-        self.description = """Outdated PHP versions were detected: {versions}.
+        self.name = "check for not supported PHP versions"
+        self.target_versions = [version.PHPVersion(ver) for ver in target_versions]
+        self.description = """Not supported PHP versions were detected: {versions}.
 \tRemove outdated PHP packages via Plesk Installer to proceed with the conversion:
 \tYou can do it by calling the following command:
 \tplesk installer remove --components {remove_arg}
 """
 
+    def _format_description(self, installed_unsupported_php: typing.List[str]) -> str:
+        return self.description.format(
+            versions=", ".join(installed_unsupported_php),
+            remove_arg=" ".join([php_version.replace(" ", "") for php_version in installed_unsupported_php]).lower()
+        )
+
     def _do_check(self) -> bool:
-        log.debug(f"Checking for minimum installed PHP version of {self.min_version}")
-        # TODO: get rid of the explicit version list
-        known_php_versions = php.get_known_php_versions()
+        log.debug(f"Checking only following PHP versions installed: {self.target_versions}")
 
-        log.debug(f"Known PHP versions: {known_php_versions}")
-        outdated_php_versions = [php for php in known_php_versions if php < self.min_version]
-        outdated_php_packages = {f"plesk-php{php.major}{php.minor}": str(php) for php in outdated_php_versions}
-        log.debug(f"Outdated PHP versions: {outdated_php_versions}")
+        installed_unsupported_php = php.get_php_versions_by_condition(
+            lambda php: php not in self.target_versions and packages.is_package_installed(f"plesk-php{php.major}{php.minor}")
+        )
 
-        installed_pkgs = packages.filter_installed_packages(outdated_php_packages.keys())
-        log.debug(f"Outdated PHP packages installed: {installed_pkgs}")
-        if len(installed_pkgs) == 0:
+        if len(installed_unsupported_php) == 0:
             log.debug("No outdated PHP versions installed")
             return True
 
-        self.description = self.description.format(
-            versions=", ".join([outdated_php_packages[installed] for installed in installed_pkgs]),
-            remove_arg=" ".join(outdated_php_packages[installed].replace(" ", "") for installed in installed_pkgs).lower()
-        )
+        self.description = self._format_description(installed_unsupported_php)
 
-        log.debug("Outdated PHP versions found")
         return False
 
 
-class AssertMinPhpVersionUsedByWebsites(action.CheckAction):
-    min_version: version.PHPVersion
+class AssertMinPhpVersionInstalled(AssertInstalledPhpVersionsInList):
```

Comment:
`AssertMinPhpVersionInstalled` shouldn't be a subclass of `AssertInstalledPhpVersionsInList`, because the former asserts $version >= min$ and the latter $version \in Allowed$. Thus, `AssertMinPhpVersionInstalled` can't fulfill the role of `AssertInstalledPhpVersionsInList`, which violates the Liskov substitution principle.

Also now `AssertMinPhpVersionInstalled` would fail encountering a future unknown version of PHP.

Reply:
Let's just use condition check as the parent of both